OpenRA.Game/Map/Map.cs

```diff
@@ -760,9 +802,8 @@ public void FixOpenAreas(Ruleset rules)
 
 		public byte GetTerrainIndex(CPos cell)
 		{
-			var uv = cell.ToMPos(this);
-			var custom = CustomTerrain[uv];
-			return custom != byte.MaxValue ? custom : cachedTileSet.Value.GetTerrainIndex(MapTiles.Value[uv]);
+			var custom = CustomTerrain[cell];
+			return custom != byte.MaxValue ? custom : cachedTileSet.Value.GetTerrainIndex(MapTiles.Value[cell]);
```

Comment:
Ok you got the point, I'll revert that function back. I hope in a future pathfinder graphs will be cached so that this function will only get called once at most per cell.